backend/archiver/flows/retrieve_datasets_flow.py

```
@@ -81,19 +95,27 @@ def retrieve_single_dataset_flow(dataset_id: str, job_id: UUID, scicat_token: Se
     ).submit(dataset_id=dataset_id, token=scicat_token, wait_for=[dataset_update])  # type: ignore
 
     # TODO: check if on retrieval bucket
+    # https://github.com/SwissOpenEM/ScopeMArchiver/issues/170
     datablocks_not_in_retrieval_bucket = datablocks.result()
 
     # TODO: check if enough space available in bucket
+    # https://github.com/SwissOpenEM/ScopeMArchiver/issues/169
     retrieval_tasks = []
     for d in datablocks_not_in_retrieval_bucket:
-        retrieval_tasks.append(copy_datablock_from_LTS_to_S3.submit(dataset_id=dataset_id, datablock=d))
+        copy_to_scratch = copy_datablock_from_LTS_to_scratch.submit(dataset_id=dataset_id, datablock=d)
+
+        verify_datablock = verify_data_on_scratch.submit(
+            dataset_id=dataset_id, datablock=d, wait_for=[copy_to_scratch]
+        )
```

What happens when verification fails? Do we retry? Or do we re-read from tape? Or do we deliver to S3 whatever we have?

---

What we do: see line 49, no retries. 
What we should do: Not quite sure. Imo this implies that something went wrong on a system level (the archival verified that the data was written correctly on the LTS). We could retry (this implies that something in the transfer/untar went wrong).

What's your opinion?

---

I think, when it is unlikely that a retry would resolve the situation, we should simply throw an error and die gracefully.

I guess when the task `verify_data_on_scratch` fails it will do just that.